app-frontend/src/app/components/tools/reclassifyEntry/reclassifyEntry.controller.js

```diff
@@ -0,0 +1,81 @@
+export default class ReclassifyEntryController {
+    constructor(reclassifyService) {
+        'ngInject';
+        this.reclassifyService = reclassifyService;
+    }
+
+    $onInit() {
+        this._classRange = this.classRange;
+        this._classValue = this.classValue;
+        this.rangeValid = !!this._classRange;
+        this.valueValid = !!this._classValue;
+        // A nonexistent entry is valid, so if we're just coming into existence with invalid
+        // values, we need to notify of a change.
+        if (!this.isValid()) {
+            this.onValidityChange({validity: false});
+        }
+    }
+
```

Comment:
$onChanges(changes){...}

Reply:
It turned out to be necessary to add `$onChanges` to the table component rather than here, because the the process of the table component translating to its internal representation means that in practice, the entries are re-instantiated rather than having their binding values altered.